src/orb.yml

```
@@ -227,3 +256,44 @@ jobs:
           bundle-key: << parameters.bundle-key >>
           deployment-config: << parameters.deployment-config >>
           deployment-group: << parameters.deployment-group >>
+
+  deploy-ecs:
+    description: Deploy application to AWS ECS
+    executor: aws-cli/default
+    parameters:
+      application-name:
+        type: string
+      environment:
+        type: string
+      repo:
+        type: string
+      tag:
+        type: string
+    steps:
+      - checkout
+      - aws-cli/setup
+      - run:
+          name: Evaluate task definition template
+          command: |
+            export APP_ENV="<< parameters.environment >>"
+            export APP_IMAGE="${AWS_ECR_ACCOUNT_URL}/<< parameters.repo >>:<< parameters.tag >>"
```

Much of a muchness, but I think this might also work
```suggestion
          environment:
            APP_ENV: << parameters.environment >>
            APP_IMAGE: "${AWS_ECR_ACCOUNT_URL}/<< parameters.repo >>:<< parameters.tag >>"
          command: |
```

---

Was worried AWS_ECR_ACCOUNT_URL wouldn't get interpolated like that, but to be fair I didn't actually test it. I'll give it a try

---

No joy, looks like it doesn't interpolate the nested AWS_ECR_ACCOUNT_URL when it's passed that way 😞

At least we know for sure

---

Just had another look in the docs... yeah, this is not a thing! My bad, sorry. It's a bit inconsistent - some keys can take env vars, but clearly not this one.